File: luis-calculator-2012-04-26-143637/lib/calculator.rb
class Calculator
  attr_accessor :expr

  def initialize(expression="")
    begin
      elements = expression.split(",")
      elements.each {|element| eval(element)}
      self.expr = expression
    rescue e
    end
  end

  def sum
    elements = expr.split(',')
    if elements.empty?
      result = 0
    else
      result = elements.reduce{|s, e| e.to_i + s.to_i }.to_i
    end
    result
  end
end

File: carl-ek-calculator-2016-08-23-153216/lib/calculator.py
class Calculator:

    def __init__(self, value='', base = 10):
        self.expression = value
        self.base = base

    def add(self):
        # add comma separated values 
        # empty string gives 0
        if len(self.expression) == 0:
            return 0
        nums = self.expression.split(',')
        sum = 0
        for n in nums:
            sum += int(n, self.base)
                
        return sum
    
